src/hubbleds/components/dotplot_viewer/dotplot_viewer.py

```diff
@@ -366,15 +366,16 @@ def on_zoom(bounds_old, bounds_new):
                 
             line_marker_at.subscribe(lambda new_val: _update_lines(value = new_val))
             vertical_line_visible.subscribe(lambda new_val: _update_lines())
-            def reset_hist_n_bin():
+            def reset_hist_n_bin(xmin = None, xmax = None):
+                print("reset_hist_n_bin")
                 if nbin_func is not None:
-                    dotplot_view.state.hist_n_bin = nbin_func(dotplot_view.state.x_min, dotplot_view.state.x_max)
+                    dotplot_view.state.hist_n_bin = nbin_func(xmin or dotplot_view.state.x_min, xmax or dotplot_view.state.x_max)
```

Comment:
I don't know whether it would ever matter for us, but just a note that this could cause problems if either `xmin` or `xmax` is zero, as zero is falsy so `0 or <X>` is always `<X>`

Reply:
This is a good point. I have resolved this be explicitly checking for None